File: AkitaModel/Courier/Channel.cs
using System;

namespace Skyware.Lis.AkitaModel.Courier;


/// <summary>
/// Communication channel.
/// </summary>
public class Channel
{

    /// <summary>
    /// PK.
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    /// Human-friendly name.
    /// </summary>
    public string Name { get; set; }

    /// <summary>
    /// Filtering parameter. If null - any schema.
    /// </summary>
    public int? SchemaId { get; set; }

    /// <summary>
    /// Filtering parameter. If null - any fund.
    /// </summary>
    public int? FundId { get; set; }

    public string FilterScript { get; set; }

    public string Dialect { get; set; }

    public object DialectParams { get; set; }

    public string Transport { get; set; }

    public object TransportParams { get; set; }

    public string Target { get; set; }

    public object TargetParams { get; set; }

    public bool IsActive { get; set; } = true;

    /// <summary>
    /// API key for HTTP GET transport.
    /// </summary>
    public string ApiKeyGet { get; set; }

    public DateTime Created { get; set; } = DateTime.UtcNow;

    /// <summary>
    /// Creation time as local date/time.
    /// </summary>
    public DateTime CreatedLocalTime => Created.ToLocalTime();

    public DateTime? Modified { get; set; }

    /// <summary>
    /// Modification time as local date/time.
    /// </summary>
    public DateTime? ModifiedLocalTime => Modified?.ToLocalTime();

}

File: AkitaMotelTS/src/Courier/Channel.ts
/**
 * Skyware.Lis.AkitaModel.Courier
 */

/**
 * Communication channel.
 */
export class Channel {
  /**
   * PK.
   */
  id!: number;

  /**
   * Human-friendly name.
   */
  name?: string;

  /**
   * Filtering parameter. If null - any schema.
   */
  schemaId?: number;

  /**
   * Filtering parameter. If null - any fund.
   */
  fundId?: number;

  filterScript?: string;

  dialect?: string;

  dialectParams?: any;

  transport?: string;

  transportParams?: any;

  target?: string;

  targetParams?: any;

  isActive: boolean = true;

  /**
   * API key for HTTP GET transport.
   */
  apiKeyGet?: string;

  created: Date = new Date();

  /**
   * Creation time as local date/time.
   */
  get createdLocalTime(): Date {
    return new Date(this.created.toLocaleString());
  }

  modified?: Date;

  /**
   * Modification time as local date/time.
   */
  get modifiedLocalTime(): Date | undefined {
    return this.modified ? new Date(this.modified.toLocaleString()) : undefined;
  }
}
